### PR TITLE
`azurerm_network_interface` - removes `Computed` from `dns_server` and renames `enable_*` to `*_enabled`

### LINKED ISSUE #21225: azurerm_network_interface dns_servers not updated when changing from defined IPs to null
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.3

### AzureRM Provider Version

3.50.0

### Affected Resource(s)/Data Source(s)

azurerm_network_interface

### Terraform Configuration Files

```hcl
resource "azurerm_network_interface" "nic" {
  name                = "test-interface"
  location            = "northeurope"
  resource_group_name = "[REDACTED]"

  dns_servers = ["8.8.8.8", "9.9.9.9"]


  ip_configuration {
    name                          = "test"
    subnet_id                     = "[REDACTED]"
    private_ip_address_allocation = "Dynamic"
  }
}
```

### Debug Output/Panic Output

1
```shell

→ terraform apply
azurerm_network_interface.nic: Refreshing state... [id=/subscriptions/[REDACTED]/resourceGroups/[REDACTED]/providers/Microsoft.Network/networkInterfaces/test-interface]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
```

2
```
sed -i 's/dns_servers/# dns_servers/' iface.tf
```

3
```
→ terraform apply
azurerm_network_interface.nic: Refreshing state... [id=/subscriptions/[REDACTED]/resourceGroups/[REDACTED]/providers/Microsoft.Network/networkInterfaces/test-interface]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
```

### Expected Behaviour

`dns_servers` attribute value should be updated when its value is modified.

### Actual Behaviour

Modifying `dns_servers` attribute from a list of IPs to `null` produces no changes in `terraform apply`

### Steps to Reproduce

1. `terraform apply` provided example configuration file filling `[REDACTED]` values to real values.
2. Comment `dns_servers` attribute in provided configuration file
3. `terraform apply` again, noticing that terraform reports `No changes. Your infrastructure matches the configuration.`

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by stephybun**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

- Removed `Computed` from `dns_servers`
- Renamed `enable_ip_forwarding` to `ip_forwarding_enabled`
- Renamed `enable_accelerated_networking` to `accelerated_networking_enabled`
- Updated the update function to patch the existing resource model instead of populating a new model

## Testing 

image

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [x] Breaking Change

## Related Issue(s)
Fixes #21225

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/network/network_interface_data_source.go [hunk 1]
```diff
@@ -177,6 +176,18 @@ func dataSourceNetworkInterface() *pluginsdk.Resource {
 			"tags": commonschema.TagsDataSource(),
 		},
 	}
+
+	if !features.FourPointOhBeta() {
+		dataSource.Schema["enable_ip_forwarding"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
```
Comment (mbfrahry) [inline internal/services/network/network_interface_data_source.go:183]: Do we want to add a Deprecated messages to this attribute?

File: internal/services/network/network_interface_data_source.go [hunk 2]
```diff
@@ -177,6 +176,18 @@ func dataSourceNetworkInterface() *pluginsdk.Resource {
...
+		}
+		dataSource.Schema["enable_accelerated_networking"] = &pluginsdk.Schema{
```
Comment (mbfrahry) [inline internal/services/network/network_interface_data_source.go:185]: And do we want to add a Deprecated messages to this attribute?

**Review by mbfrahry — APPROVED** [review]
LGTM with just a minor comment on adding Deprecated messages to the fields we are removing from the datasource

**Comment by mbfrahry** [comment]
There are also a number of tests that are using these Deprecated attributes? Should we go through those now and remove them in favor of the new attributes?

**Comment by stephybun** [comment]
Thanks @mbfrahry, replaced the deprecated properties with the new ones in the the test configs where I could.

**Pull request merged**